sdk/Sdk/ExtensionsCsprojGenerator.cs

```diff
@@ -69,13 +76,33 @@ internal string GetCsProjContent()
     </PropertyGroup>
     <ItemGroup>
         <PackageReference Include=""Microsoft.NETCore.Targets"" Version=""3.0.0"" PrivateAssets=""all"" />
-        <PackageReference Include=""Microsoft.NET.Sdk.Functions"" Version=""3.0.11"" />
+        <PackageReference Include=""Microsoft.NET.Sdk.Functions"" Version=""{netSdkVersion}"" />
         {extensionReferences}
     </ItemGroup>
 </Project>
 ";
         }
 
+        private string GetNetSdkVersion()
+        {
+            if (_azureFunctionsVersion.StartsWith("v3", StringComparison.OrdinalIgnoreCase))
+            {
+                return "3.0.11";
+            }
+
+            return "4.0.0-preview2";
+        }
+
+        private string GetTargetFrameWork()
+        {
+            if (_azureFunctionsVersion.StartsWith("v3", StringComparison.OrdinalIgnoreCase))
+            {
+                return "netcoreapp3.1";
+            }
+
+            return "net6.0";
```

Comment:
Just wondering, would it be worth us using constant strings for these version variables so they're defined in one place or nah?

Reply:
Should definitely use a constant. Will change it